library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/SessionReplayFeatureTest.kt

```diff
@@ -274,8 +281,87 @@ internal class SessionReplayFeatureTest {
+        testedFeature.stopRecording()
+        val rumSessionUpdateBusMessage = mapOf(
+            SessionReplayFeature.SESSION_REPLAY_BUS_MESSAGE_TYPE_KEY to
+                SessionReplayFeature.RUM_SESSION_RENEWED_BUS_MESSAGE,
+            SessionReplayFeature.RUM_KEEP_SESSION_BUS_MESSAGE_KEY to
+                true
+        )
+
+        // When
+        testedFeature.onReceive(rumSessionUpdateBusMessage)
+
+        // Then
+        inOrder(mockRecorder) {
+            verify(mockRecorder).registerCallbacks()
+            verify(mockRecorder).resumeRecorders()
+        }
+        verifyNoMoreInteractions(mockRecorder)
+    }
+
+    @Test
+    fun `M do nothing W rum session updated { keep false, sampled in }`() {
         // Given
+        whenever(mockSampler.sample()).thenReturn(true)
+        testedFeature.onInitialize(mockSdkCore, appContext.mockInstance)
+        testedFeature.stopRecording()
+        val rumSessionUpdateBusMessage = mapOf(
+            SessionReplayFeature.SESSION_REPLAY_BUS_MESSAGE_TYPE_KEY to
+                SessionReplayFeature.RUM_SESSION_RENEWED_BUS_MESSAGE,
+            SessionReplayFeature.RUM_KEEP_SESSION_BUS_MESSAGE_KEY to
+                false
+        )
+
+        // When
+        testedFeature.onReceive(rumSessionUpdateBusMessage)
+
+        // Then
+        verify(mockRecorder).registerCallbacks()
+        verifyNoMoreInteractions(mockRecorder)
+        verify(logger.mockInternalLogger)
+            .log(
+                InternalLogger.Level.INFO,
+                InternalLogger.Target.USER,
+                SessionReplayFeature.SESSION_SAMPLED_OUT_MESSAGE
+            )
+    }
+
+    @Test
+    fun `M do nothing W rum session updated { keep true, sampled out }`() {
+        // Given
+        whenever(mockSampler.sample()).thenReturn(false)
+        testedFeature.onInitialize(mockSdkCore, appContext.mockInstance)
+        testedFeature.stopRecording()
+        val rumSessionUpdateBusMessage = mapOf(
+            SessionReplayFeature.SESSION_REPLAY_BUS_MESSAGE_TYPE_KEY to
+                SessionReplayFeature.RUM_SESSION_RENEWED_BUS_MESSAGE,
+            SessionReplayFeature.RUM_KEEP_SESSION_BUS_MESSAGE_KEY to
+                true
+        )
+
+        // When
+        testedFeature.onReceive(rumSessionUpdateBusMessage)
+
+        // Then
+        verify(mockRecorder).registerCallbacks()
+        verifyNoMoreInteractions(mockRecorder)
+        verify(logger.mockInternalLogger)
+            .log(
+                InternalLogger.Level.INFO,
+                InternalLogger.Target.USER,
+                SessionReplayFeature.SESSION_SAMPLED_OUT_MESSAGE
+            )
+    }
+
+    @Test
+    fun `M stopRecording W rum session updated { keep false, sample in }`() {
+        // Given
+        whenever(mockSampler.sample()).thenReturn(true)
         testedFeature.onInitialize(mockSdkCore, appContext.mockInstance)
         val rumSessionUpdateBusMessage1 = mapOf(
```

Comment:
why do we need to send 2 messages?

Reply:
it was just to test the situation where the first RUM session was tracked and the next RUM session was dropped.